torch/nested/__init__.py

```diff
@@ -410,3 +411,55 @@ def nested_tensor_from_jagged(
 
     return nested_view_from_values_offsets_lengths(
         values, offsets, lengths, ragged_idx=jagged_dim, min_seqlen=min_seqlen, max_seqlen=max_seqlen)
+
+def masked_select(tensor: Tensor, mask: Tensor) -> Tensor:
+    r"""
+    Constructs a nested tensor given a a strided tensor input and a strided mask, the resulting jagged layout Nested Tensor
+    will have values equivalent to running :func:`masked_select` with the provided tensor and mask. The offsets will be 
+    generated from the contiguous True values in the mask.
```

Comment:
two things to confirm/and maybe call out in the docs?
- we'd like to allow tensor/mask that are beyond 2-D, but per the semantics of masked_select, always produce a NT that is 2-D, e.g. something like `[B, j0]`. 
- if a particular row of the NT has been completed masked out, we still keep the empty row.

Reply:
> we'd like to allow tensor/mask that are beyond 2-D, but per the semantics of masked_select, always produce a NT that is 2-D, e.g. something like [B, j0].

So this one I actually thought about a bit, right now we're essentially taking in a dense tensor of  any[n_1,n_2,..., n_m] dimensionality and a mask that is broadcastable with that tensor and we are compacting it to a single dimensional tensor since the masked_select for dense tensors is defined to produce 1D: https://pytorch.org/docs/stable/generated/torch.masked_select.html

So we're always producing something of the form [1, j0] rather than [B, j0]. We could adjust it by just retaining the first dimension if ndims > 2 and collapsing on the risk of the dimensions rather than the current behavior, but I think that moves away from what the dense version does.

Reply:
The produced NJT should have 1-D values since that is the result of applying masked_select on a strided tensor, but when that is wrapped in a NJT, the outer shape should become 2-D, i.e., [B, j0]?